lib/s3.js

```diff
@@ -141,4 +140,28 @@ module.exports = CoreObject.extend({
       });
     });
   },
+
+  listAllObjects: function(options) {
+    var client         = this._client;
+    var listObjects    = Promise.denodeify(client.listObjects.bind(client));
+    var allRevisions   = [];
+
+    function listObjectRecursively(options) {
+      return listObjects(options).then(function(response) {
+        [].push.apply(allRevisions, response.Contents);
+
+        if (response.IsTruncated) {
+          var nextMarker = response.NextMarker || response.Contents[response.Contents.length - 1].Key;
```

Comment:
I actually never saw an instance where `response.NextMarker` was set.

I implemented your suggestion in 5b15556790b5feba7f7e36540702fdf9afa55e33 but personally I find it adds noise to the code. What do you think?